### PR TITLE
print deprecationwarning if before_suite and after_suite are used

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/testsuite.py [hunk 1]
```diff
@@ -584,15 +566,20 @@ def start(
...
-        suite_log = self.__log
-        (
-            is_suite_continue,
-            suite_error_message,
-            suite_error_stacktrace,
-        ) = self.__suite_method(
-            self.before_suite, test_kwargs=test_kwargs, log=suite_log
-        )
+        if hasattr(self, "before_suite"):
+            suite_log = self.__log
```
Comment (squirrelsc) [inline lisa/testsuite.py]: the suite_log is used in other places, it should be initialized always.

Comment (LiliDeng) [reply lisa/testsuite.py]: updated

Comment (squirrelsc) [reply lisa/testsuite.py]: This line should be removed now.

Comment (LiliDeng) [reply lisa/testsuite.py]: removed

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**